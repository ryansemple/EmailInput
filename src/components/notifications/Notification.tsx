import React from "react";
import clsx from "clsx";

interface NotificationProps {
	text: string,
	className: string
}

/**
 * Renders a notification.
 */
const Notification = (props: NotificationProps) => (
	<div 
		className={clsx("Notifications_Notification", props.className)} 
		role="alert">
		<p>{props.text}</p>
	</div>
)

export default Notification;