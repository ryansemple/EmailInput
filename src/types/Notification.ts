import ObjectWithId from "../types/ObjectWithId";

export enum NotificationType {
	Success,
	Error
}

export interface NotificationInstance extends ObjectWithId {
	text: string,
	id: string,
	notificationType: NotificationType
}