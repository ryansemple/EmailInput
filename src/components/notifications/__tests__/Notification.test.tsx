import React from "react";
import { render } from "@testing-library/react";
import Notification from "../Notification";

test("Renders Notification component correctly.", () => {
	const { container } = render(
		<Notification 
			text="Notification"
			className="class"
		/>
	);
	expect(container).toMatchSnapshot();
});