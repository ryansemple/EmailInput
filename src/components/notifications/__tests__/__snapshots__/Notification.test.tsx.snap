// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Notification component correctly. 1`] = `
<div>
  <div
    class="Notifications_Notification class"
    role="alert"
  >
    <p>
      Notification
    </p>
  </div>
</div>
`;
